vagrant.yml

```diff
@@ -0,0 +1,13 @@
+---
+box: centos/7
+
+# Cluster VMs' ip, hostname, domain, cpus and memory (generates VM deployment config)
+hosts:
+  # |------ip-------|-hostname-|domain|cpus|memory|
+  - ["192.168.56.10", edge-01, tdp, 2, 2048]
+  - ["192.168.56.11", master-01, tdp, 3, 3072]
+  - ["192.168.56.12", master-02, tdp, 3, 3072]
+  - ["192.168.56.13", master-03, tdp, 3, 3072]
+  - ["192.168.56.14", worker-01, tdp, 2, 2048]
+  - ["192.168.56.15", worker-02, tdp, 2, 2048]
+  - ["192.168.56.16", worker-03, tdp, 2, 2048]
```

Comment:
Wouldn't it be more pertinent if the host list was something like:
```yaml
hosts:
  masters:
    - ip: ...
      hostname: ...
      domain: ... # not sure if setting the domain for each is pertinent either
      ...
  workers:
    - ...
  edges:
    - ...
```
(hosts can also be a flat list of dicts (without masters, workers and edges))


Hash map destructuration is possible with:
```ruby
hash = {:a => 1, :b => 2, :c => 3}
a, b = hash.values_at(:a, :b)
```

Reply:
Yes but I keep the actual host list to simplify the PR. We can update the host list structure in another PR. The goal for this PR is to experiment libvirt provider and remove the use of Ansible for generating the Vagrantfile and the inventory.

Reply:
Right, thanks for your explanation.